traitsui/wx/tabular_editor.py

```diff
@@ -937,7 +937,7 @@ def save_prefs(self):
         """
         cws = self._cached_widths
         if cws is not None:
-            cws = [(None, cw)[cw >= 0] for cw in cws]
```

Comment:
ohh wow that took a few seconds to decipher.

Reply:
Just so i understand correctly, `cw` can be positive, negative or zero.

If `cw` is None or < 0, `[cw >= 0]` will return `[False]` which is equivalent to `[0]` on Python 2. So, if `cw` is `None` or `< 0`, we will choose the zeroth element in the tuple i.e. `None`

when `cw >= 0`, we get `[True]` which is equivalent to `[1]` on Python 2 so we choose the first element in the tuple i.e. `cw`

Reply:
Yep